### PR TITLE
update dependencies including github3.py

### PR DESCRIPTION
**Pull request by davisagli**
Also includes some tidying of github actions configuration

# Critical Changes

# Changes

# Issues Closed

### CODE REVIEW

**Comment by davisagli** [comment]
Looking at the tests that broke...

File: setup.py [hunk 1]
```diff
@@ -33,13 +33,15 @@ def find_packages(path=["."], prefix=""):
             continue
         else:
             req = req.split(" ")[0]
+            # Work around normalized name of github3.py distribution
+            req = req.replace("github3-py", "github3.py")
```
Comment (jstvz) [inline setup.py:37]: Huh. Figured setuptools/wheel would do the right thing here.

File: requirements/prod.txt [hunk 2]
```diff
@@ -30,18 +30,15 @@ cryptography==36.0.1
...
     # via -r requirements/prod.in
 faker==12.0.0
     # via
     #   -r requirements/prod.in
     #   snowfakery
-fs==2.4.14
+fs==2.4.15
     # via -r requirements/prod.in
-github3.py==3.0.0
+github3-py==3.1.0
```
Comment (jstvz) [inline requirements/prod.txt:41]: 😑

**Review by jstvz — APPROVED** [review]

File: requirements/prod.txt (hunk 2 above)
Comment (davisagli) [reply requirements/prod.txt:41]: I think it's from this change in pip-tools: https://github.com/jazzband/pip-tools/pull/1572

Comment (davisagli) [reply requirements/prod.txt:41]: I think it's from this change in pip-tools: https://github.com/jazzband/pip-tools/pull/1572

File: setup.py (hunk 1 above)
Comment (davisagli) [reply setup.py:37]: Yeah -- I think there's probably a bug somewhere in there, but I hit my timebox for locating it and decided to do this instead.

**Pull request merged**